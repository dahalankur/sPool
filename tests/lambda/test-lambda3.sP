# nested lambdas with arrow types

lambda (quack -> (quack -> string)) ():
    lambda (quack -> string) (): 
        lambda string ():
            return "hello there";;;
