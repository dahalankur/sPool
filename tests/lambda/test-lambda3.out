PROGRAM {[ 
    LAMBDA(ARROWTY(QUACKTY -> ARROWTY(QUACKTY -> STRINGTY)), FORMALS(), [ 
        LAMBDA(ARROWTY(QUACKTY -> STRINGTY), FORMALS(), [ 
            LAMBDA(STRINGTY, FORMALS(), [ 
                RETURN(STRING("hello there"))])])])]}
